c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -641,41 +705,79 @@ class DeviceCachingAllocator {
     p.block = new Block(p.device(), p.stream(), size, p.pool, (char*)ptr);
     update_stat_array(stats.segment, 1, p.stat_types);
     update_stat_array(stats.reserved_bytes, size, p.stat_types);
+    if (size >= s_config.max_split_size)
+      update_stat(stats.oversize_segments, 1);
 
     return (p.block != nullptr);
   }
 
-  bool free_cached_blocks()
+  /** Free one or more blocks to the system allocator.  But only enough to satasfy the target size **/
+  bool release_available_cached_blocks(const AllocParams& p)
```

Comment:
This function seems to be called no matter if splitting is turned on or off. Is this intentional? Is this optimization helpful even in normal OOMs?